### PR TITLE
Site Editor: Organise semantic template parts

### LINKED ISSUE #27877: Site Editing: In the W navigation, utilise drilldown pattern to organise semantic template parts
Once we have semantic template part categorisation in place, we can utilise the drilldown pattern in the W navigation to better organise template parts therein:

Screenshot 2020-12-23 at 13 44 01

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Fixes https://github.com/WordPress/gutenberg/issues/27877
* Create sub-menus inside the Template Parts menu for Headers, Footers, and Sidebars.
* Put the rest of the template parts into General.
* Empty menus are hidden.

## How has this been tested?
1. Open site editor
2. Open browsing sidebar
3. Click on template parts
4. You should see Headers if you have header template parts
5. You should see Footers if you have footer template parts
6. You should see Sidebars if you have sidebar template parts
7. You should see General if you have any other templates parts that don't fit into headers, footers, and sidebars

## Screenshots 
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by jameskoster — APPROVED** [review]
It works! :)

**Review by aristath — APPROVED** [review]

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js [hunk 1]
```diff
@@ -30,3 +35,35 @@ export const MENU_TEMPLATES_PAGES = 'templates-pages';
...
+		menu: MENU_TEMPLATE_PARTS_FOOTERS,
+		title: __( 'Footers' ),
+	},
+	{
+		area: TEMPLATE_PART_AREA_SIDEBAR,
+		menu: MENU_TEMPLATE_PARTS_SIDEBARS,
+		title: __( 'Sidebars' ),
+	},
+	{
+		area: null,
```
Comment (vindl) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js:65]: Aligning this with changes in @Addison-Stavlo PR https://github.com/WordPress/gutenberg/pull/28410, would it make sense to assign `uncategorized` here too?

**Review by vindl — COMMENTED** [review]
Considering that most of the sites will likely have just one template part per area, one downside of organizing things this way is that it introduces one more click to get to the menu with only one item. On the other hand, it better communicates what template parts are and their relationship with templates so it might be worthwhile to try this out.

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js (hunk 1 above)
Comment (david-szabo97) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js:65]: `null` includes all template parts from the areas that aren't specified in this constant array. In this case template parts with `header`, `footer` or `sidebar` will be excluded from **General** but everything else will be included. This only matters if a theme has an area besides the header,footer and sidebar. So basically `null` is a catch-all, excluding items from the three other areas.

Comment (vindl) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js:65]: What I was trying to say is that is that we are using `uncategorized` for the same thing (catch-all) on back end.

Comment (david-szabo97) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js:65]: Just checked the code you are right. I thought we allow other areas as well. I'll update the code to use `uncategorized`

**Review by vindl — APPROVED** [review]

**Pull request merged**